### DISCUSSION TITLE
Build docker image targeting multiple architures

### DISCUSSION THREAD

**Discussion by fsheik** [outline/outline#2255] [Ideas]
I am trying to setup outlinewiki on docker installed over a Raspberry Pi (RPi) 4. Since the current docker image targets only linux/amd64, it cannot be used with RPi. How about we modify the build pipeline to generate linux/arm64 image as well?

**Comment by tommoor** [comment]
We'd be open to it – it previously came up here with some idea of what would be required:

https://github.com/outline/outline/discussions/1936#discussioncomment-774662

**Reply by fsheik** [reply]
Thanks Tom! Approach 2 in [this](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/) blog post looks neat. Would you be able to configure DOCKER_PASSWORD and DOCKER_USERNAME in circleci environment variables? Post that we can update config.yml to perform docker login and push conditionally (i.e. only when we want to publish a new version/official build from master). [Here's](https://github.com/jdrouet/docker-on-ci/blob/master/.circleci/config.yml) an example of how config.yml will look like.

**Reply by tommoor** [reply]
Currently this is is configured on the side of DockerHub – it's just config in their UI rather than stored in the repo. I think the bulk of the work here though is creating a cross platform dockerfile?

**Reply by fsheik** [reply]
Simpler approach would be if we integrate this step (see screenshot below from [here](https://www.docker.com/blog/multi-arch-build-and-images-the-simple-way/)) as part of a circleci pipeline and push to Docker Hub. When we do this we will have to go away from using the UI based configuration on DockerHub since we would be pushing a build as part of circleci build.

I will try out a working example with my forked branch and circleci pipleine next week and let you know.

![image](https://user-images.githubusercontent.com/4527258/123860393-7726cd80-d943-11eb-97a9-cebe0fd0fd4f.png)

**Reply by tommoor** [reply]
By all means have a go, I'm not convinced it's simpler – just different.
